src/main/scala/org/ergoplatform/mining/ProofOfUpcomingTransactions.scala

```diff
@@ -0,0 +1,39 @@
+package org.ergoplatform.mining
+
+import io.circe.{Encoder, Json}
+import org.ergoplatform.modifiers.history.{BlockTransactions, HeaderSerializer, HeaderWithoutPow}
+import org.ergoplatform.nodeView.mempool.TransactionMembershipProof
+import org.ergoplatform.settings.Algos
+import io.circe.syntax._
+
+/**
+  * Proof of inclusion of certain transactions into a block with known and yet unproven header
```

Comment:
please describe the rationale/motivation behind this design, what are the goals, etc.

Reply:
A particular example is provided.